alerters/alerters-plugins/alerters-kafka/src/main/java/org/hawkular/alerter/kafka/KafkaAlerter.java

```diff
@@ -0,0 +1,251 @@
+ *
+ *   Kafka Consumer properties are prefixed with "kafka." in the Trigger context.
+ *   kafka.key.deserializer and kafka.value.deserializer point to StringDeserializer as default if not present.
+ *
+ *   i.e.   kafkaTrigger.addContext("kafka.bootstrap.servers", "localhost:9092");
+ *          kafkaTrigger.addContext("kafka.group.id", "kafka-trigger-group");
+ *          kafkaTrigger.addContext("kafka.key.deserializer", StringDeserializer.class.getName());
+ *          kafkaTrigger.addContext("kafka.value.deserializer", StringDeserializer.class.getName());
+ *
+ *   See the KafkaConsumer reference for more info
+ *
+ *   https://kafka.apache.org/0100/javadoc/index.html?org/apache/kafka/clients/consumer/KafkaConsumer.html
+ *
+ * - [Required]    trigger.context["topic"] = "<kafka topic to listen>"
+ *
+ *   It defines the Kafka Topic a trigger will listen. In the current version only a single topic for Trigger is allowed.
+ *
+ * - [Optional]    trigger.context["poll_timeout"] = "<kafka consumer poll timeout>"
+ *
+ *   It defines the poll timeout in ms for the Kafka Consumer. By default it takes 1 second.
+ *
+ * - [Optional]    trigger.context["mapping"] = "<mapping_expression>"
+ *
+ *   By default, Kafka records are directly mapped into a Data object, with the following mapping
+ *
+ *          Kafka record timestamp -> Data timestamp
+ *          Kafka record value     -> Data value
+ *          Kafka Topic            -> Data id (dataId referenced on conditions)
+ *
+ *   Optionally, a Kafka record can be mapped into an Event. To enable this, a "mapping" expression must be present in
+ *   Trigger context.
+ *
+ *   Kafka Alerter expects a json record value.
+ *
+ *   A mapping expressions defines how to convert an Kafka json record into a Hawkular Event:
+ *
+ *   <mapping_expression> ::= <mapping> | <mapping> "," <mapping_expression>
+ *   <mapping> ::= <kafka_record_field> [ "|" "'" <DEFAULT_VALUE> "'" ] ":" <hawkular_event_field>
+ *   <hawkular_event_field> ::= "id" | "ctime" | "dataSource" | "dataId" | "category" | "text" | "context" | "tags"
+ *
+ * - [Optional]    trigger.context["timestamp_pattern"] = "<date and time pattern>"
+ *
+ *   By default, an Event ctime field is mapped directly from a kafka record timestamp value. This can be optionally
+ *   overwritten to use a json field of the Kafka record value.
+ *
+ *   This property defines a time pattern to parse ctime field. It must follow supported formats of
+ *   {@link java.time.format.DateTimeFormatter}.
+ *
+ * </pre>
+ *
+ * @author Jay Shaughnessy
+ * @author Lucas Ponce
+ */
+@Alerter(name = "kafka")
+public class KafkaAlerter implements AlerterPlugin {
+    private static final MsgLogger log = MsgLogging.getMsgLogger(KafkaAlerter.class);
+
+    private static final String KAFKA_ALERTER = "hawkular-alerts.kafka-alerter";
+    private static final String KAFKA_ALERTER_ENV = "KAFKA_ALERTER";
+    private static final String KAFKA_ALERTER_DEFAULT = "true";
```

Comment:
I think we commented some weeks ago that we can enable them by default.
That's good for community as less steps to test the project.
I think there is space for performance improvements and yes, at any sign that is creating some overhead we can disable it in the config.
As our goal is to impact in community in short term, I think is good to maintain them enabled (all of them, elasticsearch, prometheus, kafka and others).